### DISCUSSION TITLE
Can the docker image be used without X11 server?

### DISCUSSION THREAD

**Discussion by sachinlionel** [firelab/windninja#492] [General]
We are looking for an automated use of this service, is there a way to use it without GUI.

**Comment by nwagenbrenner** [comment]
Yes, you could modify the docker build to build without the GUI and use the
command line interface. You'll need to set the NINJA_QTGUI option to OFF in
CMake.

**Reply by sachinlionel** [reply]
thanks, will try it out.

**Comment by nwagenbrenner** [comment]
Great, just let us know if you have questions or run into problems. Please let us know if you get something working.

**Comment by sachinlionel** [comment]
Addition to NINJA_QTGUI option to OFFm I had to do some corrections in docker file to get this running. 

![Screenshot 2023-06-26 at 11 32 20](https://github.com/firelab/windninja/assets/25305931/264748d9-5d4c-46b8-92f1-b45c415d8ae3)

```
RUN cd  /opt/src/windninja && \
    mkdir build && \
    cd /opt/src/windninja/build && \
    cmake -D SUPRESS_WARNINGS=ON -D NINJAFOAM=ON .. && \
    make -j4 && \
    make install && \
    ldconfig && \
    cd /opt/src/windninja && \
    /usr/bin/bash -c scripts/build_libs.sh

CMD /usr/bin/bash -c /usr/local/bin/WindNinja_cli
VOLUME /data
WORKDIR /data
```

**Comment by nwagenbrenner** [comment]
Okay, great. Glad you got it working. Thanks for letting us know. I suppose we could provide multiple Dockerfiles for different builds (e.g., with an without the GUI).

**Reply by sachinlionel** [reply]
that would be helpful. 
Or a script which wrappers the ENV variable for different (non/-gui) builds.

**Comment by ghost** [comment]
hi @sachinlionel  Maybe you can simply override the entrance point when running the Docker instead of modifying the Dockerfile and having different images. Something like:

```
computeadmin@tsylHPC1:~$ sudo docker run -it --rm -v /home/computeadmin/projects/tests/docker2/example-files:/home/wind/example  windninja:3.9 /usr/bin/bash -c 'WindNinja_cli /home/wind/example/cli_domainAverage.cfg'
Run 0: Reading elevation file...
Run 0: Run number 0 started with 1 threads.
Run 0: Generating mesh...
Run 0: Initializing flow...
Run 0: Building equations...
Run 0: Solving...
Run 0 (solver): 14% complete
Run 0 (solver): 37% complete
Run 0 (solver): 60% complete
Run 0 (solver): 81% complete
Run 0 (solver): 92% complete
Run 0 (solver): 99% complete
Run 0 (solver): 100% complete
Run 0: Writing output files...
Run 0: Meshing time was 0.015796 seconds.
Run 0: Initialization time was 0.020119 seconds.
Run 0: Equation building time was 0.317765 seconds.
Run 0: Solver time was 1.490952 seconds.
Run 0: Output writing time was 0.133942 seconds.
Run 0: Total simulation time was 2.121692 seconds.
Run 0: Run number 0 done!
computeadmin@tsylHPC1:~$
```
hope this helps

**Comment by sachinlionel** [comment]
Thanks, yeah it can be done this way as well.
I had to modify dockerfile to build non-gui anyways, so prefer to use change CMD in dockerfile.
